### PR TITLE
full-heal: several fixes

### PR DESCRIPTION
**Pull request by AtomicChicken**
- fix: clothed units no longer exhibit melting body parts after being healed
- fix: resetting unit.status2.limbs_fly_count allows wounded creatures to fly again
- added several missing counters and flags
- added capacity to clear wounds stored in historical figure data
- removed the buggy CleanSelf job assignment

### CODE REVIEW

File: full-heal.lua [hunk 1]
```diff
@@ -162,9 +165,21 @@ if unit then
         status.severed_or_jammed = false
     end
 
-    for i, temp in ipairs(unit.status2.body_part_temperature) do
-        local bp = unit.body.body_plan.body_parts[i]
-        temp.whole = math.floor((bp.min_temp + bp.max_temp) / 2)
+    for i = #unit.status2.body_part_temperature-1,0,-1 do
+      unit.status2.body_part_temperature:erase(i) -- attempting to rewrite temperature was causing body parts to melt for some reason; forcing repopulation in this manner appears to be safer
```
Comment (lethosor) [inline full-heal.lua]: Any idea what this was setting the temperature to? Before I added it, body parts of creatures that had died by melting would *always* melt.

Comment (AtomicChicken) [reply full-heal.lua]: Can't say I know. Your method sets body part temperatures to what are initially safe levels, but they proceed to rise abnormally in a gradual manner until melting point is reached; this appears to be especially likely for clothed (insulated) body parts.

Comment (lethosor) [reply full-heal.lua]: Ah, okay, it's possible I didn't notice that. If this also works for melted and resurrected creatures, it's fine with me.

Comment (AtomicChicken) [reply full-heal.lua]: A few tests later, I can say that resurrecting creatures after having submerged them in magma appears to work without issue.

File: full-heal.lua [hunk 2]
```diff
@@ -148,16 +161,68 @@ if unit then
...
     end
 
-    if unit.job.current_job and unit.job.current_job.job_type == df.job_type.Rest then
-        --print("Wake from rest -> clean self...")
-        unit.job.current_job.job_type = df.job_type.CleanSelf
+    for i = 0,#unit.enemy.body_part_8a8-1,1 do
+      unit.enemy.body_part_8a8[i] = 1 -- not sure what this does, but values appear to change following injuries
```
Comment (lethosor) [inline full-heal.lua]: What exactly are following injuries? Injuries related to this specific body part?

**Comment by lethosor** [comment]
If you have any idea what the unknown stuff is, or figure it out later, it would be great to get that into structures. Otherwise, let me know when this is merge-ready.

File: full-heal.lua (hunk 2 above)
Comment (AtomicChicken) [reply full-heal.lua]: Affirmative for 'injuries related to this specific body part', but I really don't know anything beyond that. To give a tangible example: severing a body part off changes the corresponding value (indexed by body part id) in body_part_878 to 0. Same for body_part_8a8 and body_part_888.

**Comment by AtomicChicken** [comment]
I _think_ that covers everything; you can probably go ahead and merge this.

**Comment by lethosor** [comment]
If you have a good list of things to add to the changelog (also for #52?), that would be appreciated. Otherwise I'll try to do that when I get around to updating the DFHack repo.

**Pull request merged**